src/admin/pages/Login.tsx

```diff
@@ -0,0 +1,99 @@
+import { ChangeEvent, FormEvent, useState } from "react";
+
+import { useLocalStorage } from "../../hooks/useLocalStorage";
+import { loginAdmin } from "../../network/adminApi";
+import { Jwt } from "../../network/jwt";
+import { FetchError } from "../../network/request";
+import styles from "./Login.module.scss";
+
+interface LoginProps {
+  onLogin(jwt: Jwt): void;
+}
+
+export const Login = ({ onLogin }: LoginProps) => {
+  const [rememberUsername, setRememberUsername] = useLocalStorage<boolean>(
+    "admin.rememberUsername",
+    true
+  );
+  const [username, setUsername] = useLocalStorage<string>(
+    "admin.cacheUsername",
+    ""
+  );
+  const [password, setPassword] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  function changeUsername(e: ChangeEvent<HTMLInputElement>) {
+    const username = e.target.value;
+    setUsername(username, { remember: rememberUsername });
+    setErrorMessage("");
+  }
+
+  function changePassword(e: ChangeEvent<HTMLInputElement>) {
+    const password = e.target.value;
+    setPassword(password);
+    setErrorMessage("");
+  }
+
+  function changeRemember(e: ChangeEvent<HTMLInputElement>) {
+    const remember = e.target.checked;
+    setRememberUsername(remember);
+    setUsername((v) => v, { remember });
+  }
+
+  function submit(e: FormEvent) {
+    e.preventDefault();
+
+    if (username === "") {
+      setErrorMessage("Error: Username missing");
+    } else if (password === "") {
+      setErrorMessage("Error: Password missing");
+    } else {
+      attemptLogin();
+    }
+  }
+
+  async function attemptLogin() {
+    try {
+      const jwt = await loginAdmin({ username, password });
```

Comment:
network requests should be done via the store and use the error handling via the store